command/kv_delete.go

```diff
@@ -96,22 +115,54 @@ func (c *KVDeleteCommand) Run(args []string) int {
 		return 2
 	}
 
-	path := sanitizePath(args[0])
-	mountPath, v2, err := isKVv2(path, client)
-	if err != nil {
-		c.UI.Error(err.Error())
-		return 2
+	// If true, we're working with "-mount=secret foo" syntax.
+	// If false, we're using "secret/foo" syntax.
+	mountFlagSyntax := (c.flagMount != "")
```

Comment:
I've been thinking how to refactor this a bit to reduce the code duplication. This is what I have so far, not sure if it will work, so feel free to ignore:

```go
func splitPaths(...) (mountPath, partialPath, fullPath string) {
	if c.flagMount != "" {
		// In this case, this arg is the secret path (e.g. "foo").
		partialPath = sanitizePath(args[0])
		mountPath = sanitizePath(c.flagMount)
		fullPath = path.Join(mountPath, partialPath)
	} else {
		// In this case, this arg is a path-like combination of mountPath/secretPath.
		// (e.g. "secret/foo")
		partialPath = sanitizePath(args[0])
		mountPath = ""
                fullPath = partialPath
	}

	return mountPath, partialPath, fullPath 
}
```

Then in the function code:

```go
      mountPath, partialPath, fullPath := splitPaths(...)
      
      mountPath, v2, err := isKVv2(path, client)
      if err != nil {
           c.UI.Error(err.Error())
           return 2
       }
 ```

Reply:
Thanks for thinking about it! Unfortunately, the "fullPath" isn't so simple. fullPath is only equal to partialPath when it's KV v1, so we need the isKVv2 call to happen first to determine fullPath. Because of that I tried squishing the isKVv2 call into something like the splitPaths function before, but that failed me when it came to some of the other KV commands, which require fullPath to come from a different kind of logic. (Fun fact: There's more than just `/data` and `/metadata`.... see KV Delete, Destroy, and Undelete!)

I think I will go with what I've got for now and if something flexible suddenly comes to mind we can make a new PR for that.